trio/_core/tests/test_run.py

```diff
@@ -1853,18 +1857,12 @@ async def __anext__(self):
             items = [None] * len(nexts)
             got_stop = False
 
-            def handle(exc):
-                nonlocal got_stop
-                if isinstance(exc, StopAsyncIteration):
-                    got_stop = True
-                    return None
-                else:  # pragma: no cover
-                    return exc
-
-            with _core.MultiError.catch(handle):
+            try:
                 async with _core.open_nursery() as nursery:
                     for i, f in enumerate(nexts):
                         nursery.start_soon(self._accumulate, f, items, i)
+            except ExceptionGroup as excgroup:
+                got_stop = bool(excgroup.split(StopAsyncIteration))
```

Comment:
split() returns a 2-tuple so this will always be true...

Reply:
It now turns the first item of the tuple into a boolean.